actions/create_appliance.py

```diff
@@ -16,32 +18,32 @@ def create_appliance(api_auth, parameters, contexts):
     :rtype: string
     """
     try:
-        city = parameters["City"]
-        site_type = parameters["SiteTypes"]
+        city = parameters["City"].replace(" ", "")  # .replace() is for locations where there are spaces. E.g. Kuala Lumpur
+        site_name = parameters["SiteName"]
         model = parameters["Model"]
-
+        country_code = parameters["Country"]["alpha-2"]
+ 
     except KeyError as e:
         error_string = "Error processing create Appliance intent. {0}".format(e)
         logging.error(error_string)
         return error_string
 
     # Get all sites and check whether site exists
+    # Currently, if site doesn't exist it just says not available
+    # Need to fix it, but depends on how get_site_id_by_name is impleented 
+    site_id = get_site_id_by_name(api_auth, site_name, city,country_code)
```

Comment:
Should this catch util.APIError exceptions and return the error string if there is one? ie.

```python
from util import APIError

[...]

try:
    site_id = get_site_id_by_name(api_auth, site_name, city,country_code)
except APIError as e:
    return str(e)
```